packages/admin-panel/src/VizBuilderApp/components/DataLibrary/SelectedOptionWithJsonEditor.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import DownArrow from '@material-ui/icons/ArrowDropDown';
+import styled from 'styled-components';
+import { BaseSelectedOption, FlexSpaceBetween, JsonEditor } from '@tupaia/ui-components/';
+
+const FlexBetweenPanel = styled(FlexSpaceBetween)`
+  width: 100%;
+`;
+
+const JsonEditorPanel = styled.div`
+  flex: 1;
+  display: flex;
+  width: 100%;
+
+  > div {
+    width: 100%;
+  }
+
+  .jsoneditor {
+    border: none;
+    cursor: text;
+  }
+`;
+
+const OptionPanelWithJsonEditor = styled.div`
+  display: flex;
+  justify-content: space-between;
+  flex-direction: column;
+  align-items: flex-start;
+  height: auto;
+  flex: 1;
+  min-width: 0;
+  overflow: hidden;
+`;
+
+const DownArrowIconWrapper = styled.div`
+  display: flex;
+
+  .icon-wrapper {
+    cursor: pointer;
+  }
+  .MuiSvgIcon-root {
+    transition: transform 0.3s ease;
+    transform: rotate(${({ $expanded }) => ($expanded ? '0deg' : '-90deg')});
```

Comment:
What's the `$` doing here? I've never seen that syntax 😮

Reply:
`styled-component` will prevent `$expanded` from being passed to DOM, which I think it is a good practice. Sometimes it will produce some warning annoyingly in console for those without $. 
Reference: https://styled-components.com/docs/api#transient-props